packages/grid/x-data-grid-pro/src/tests/exportPrint.DataGrid.test.tsx

```diff
@@ -0,0 +1,262 @@
+import * as React from 'react';
+import { DataGrid, GridToolbarExport, GridRowsProp, GridColumns } from '@mui/x-data-grid';
+import { createRenderer, screen, fireEvent } from '@mui/monorepo/test/utils';
+import { expect } from 'chai';
+import { spy } from 'sinon';
+
+describe('<DataGridPro /> - Export - Print', () => {
+  const { render, clock } = createRenderer({ clock: 'fake' });
+
+  const rows: GridRowsProp = [
+    { id: 1, col1: '1-1', col2: '1-2' },
+    { id: 2, col1: '2-1', col2: '2-2' },
+  ];
+  const columns: GridColumns = [{ field: 'col1' }, { field: 'col2' }];
+  const baselineProps = {
+    columns,
+    rows,
+    // A hack to remove the warning on print
+    rowsPerPageOptions: [2, 100],
+  };
+
+  const allBooleanConfigurations = [
+    {
+      printVisible: true,
+      gridVisible: true,
+    },
+    {
+      printVisible: false,
+      gridVisible: true,
+    },
+    {
+      printVisible: true,
+      gridVisible: false,
+    },
+    {
+      printVisible: false,
+      gridVisible: false,
+    },
+  ];
+  // We need `createObjectURL` to test the downloaded value
+  before(function beforeHook() {
+    if (/jsdom/.test(window.navigator.userAgent)) {
+      // Need layouting
+      this.skip();
+    }
+  });
+
+  describe('restore visibility', () => {
+    allBooleanConfigurations.forEach(({ printVisible, gridVisible }) => {
+      it(`should set and restore 'col1' from ${printVisible ? "'visible'" : "'hidden'"} to ${
+        gridVisible ? "'visible'" : "'hidden'"
+      } (visibilityModel)`, async function test() {
+        const onColumnVisibilityModelChange = spy();
+
+        render(
+          <div style={{ width: 300, height: 300 }}>
+            <DataGrid
+              {...baselineProps}
+              onColumnVisibilityModelChange={onColumnVisibilityModelChange}
+              components={{
+                Toolbar: () => (
+                  <GridToolbarExport
+                    printOptions={{ fields: printVisible ? ['col1', 'col2'] : ['col2'] }}
+                  />
+                ),
+              }}
+              initialState={{
+                columns: {
+                  columnVisibilityModel: {
+                    // Hide columns status and traderName, the other columns will remain visible
+                    col1: gridVisible,
+                    col2: false,
+                  },
+                },
+              }}
+            />
+          </div>,
+        );
+        expect(onColumnVisibilityModelChange.callCount).to.equal(0);
+
+        // Trigger print
+        fireEvent.click(screen.queryByRole('button', { name: 'Export' }));
+        fireEvent.click(screen.queryByRole('menuitem', { name: 'Print' }));
+        await clock.runToLast();
```

Comment:
Two things:

- `clock.runToLast` doesn't return a promise. You can use a hack to wait for the promise:
https://github.com/mui/mui-x/blob/acb00d2005b3855de60b82e8b39d3325f8b21b50/packages/grid/x-data-grid-pro/src/tests/cellEditing.DataGridPro.test.tsx#L817
- Do we need to mock the time here? Replace `nativeSetTimeout` with `setTimeout` if the time is real in the above hack.

Reply:
Ok, then I don't know why adding the wait makes tests pass

As a solution, I propose to use the `apiRef.current.exportDataAsPrint()` instead of clicking such that I can put an `await` to be sure the method exit

I add another test, to verify that the method receive the correct parameters when Print button is clicked

Reply:
I do not manage to listen at the `apiRef.current.exportDataAsPrint` using spy, so I can not verify it receive the correct arguments